packages/react-storage/src/components/StorageBrowser/createStorageBrowser.tsx

```diff
@@ -61,6 +65,12 @@ export function createStorageBrowser<
     region,
   } = input.config;
 
+  setUserAgent({
+    componentName: 'StorageBrowser',
+    packageName: 'react-storage',
+    version: VERSION,
+  });
```

Comment:
Should probably happen in the adapters rather than in the factory?

Reply:
I think factory is ok here. We need to add custom user agent if customers roll their own adapters.